Write phase 3 articles (#8)
CoherentLabs/StarterGuide: src/content/docs/phase-3-layout-assets-and-styling/Graphics & Shapes/svg-ui-tricks.mdx

</video>

:::note[Why It Works]

The drag calculation is the bridge between the two coordinate systems. `deltaX` and `deltaY` come from the mouse in screen pixels, while `panX` and
`panY` are SVG coordinates. Dividing the visible SVG window by the rendered element size converts one screen pixel into the correct amount of SVG map
movement. This is why the same interaction continues to work regardless of the map's UI size.

You are not scaling or translating the DOM node itself. You are changing the visible coordinate window inside the SVG. The map can remain a fixed-size
UI component while the internal vector coordinate system handles the camera movement.

This also avoids maintaining multiple raster map resolutions for different zoom levels. The SVG paths remain vector data, so the map stays
resolution-independent compared to texture-based minimap images.

:::

::::caution[Complex SVG Cost]

Changing `viewBox` on a complex SVG can still trigger redraw work. This is usually appropriate for controlled interactions, such as opening a map
screen or dragging a minimap, but you should avoid unnecessary **per-frame** updates on very large SVG maps. If the SVG contains many moving paths,

> labels, and decorative layers, <Highlight>profile the interaction in the target UI</Highlight>.

Reviewer: Maybe? Or something similar? `profile the interaction in the target UI` is kind of weird for me :D 

```suggestion
labels, and decorative layers, <Highlight>make sure to profile the performance of the UI</Highlight>.
```
